ElementX/Sources/Services/Media/MediaUploadingPreprocessor.swift

```diff
@@ -0,0 +1,416 @@
+                                          mimetype: result.mimeType,
+                                          size: videoSize,
+                                          thumbnailInfo: thumbnailInfo,
+                                          thumbnailSource: nil,
+                                          blurhash: thumbnailResult.blurhash)
+                
+                let mediaInfo = MediaInfo.video(videoURL: result.url, thumbnailURL: thumbnailResult.url, videoInfo: videoInfo)
+                
+                return .success(mediaInfo)
+            case .failure(let error):
+                return .failure(.failedProcessingVideo(error))
+            }
+        case .failure(let error):
+            return .failure(.failedProcessingVideo(error))
+        }
+    }
+    
+    /// Prepares a file for upload.
+    /// - Parameters:
+    ///   - url: The audio URL
+    ///   - mimeType: the mimeType extracted from the UTType
+    /// - Returns: Returns a `MediaInfo.audio` containing the file URL plus the corresponding `AudioInfo`
+    private func processAudio(at url: URL, mimeType: String?) async -> Result<MediaInfo, MediaUploadingPreprocessorError> {
+        let fileSize = try? UInt64(FileManager.default.sizeForItemAt(url))
+        
+        let asset = AVURLAsset(url: url)
+        guard let durationInSeconds = try? await asset.load(.duration).seconds else {
+            return .failure(.failedProcessingAudio)
+        }
+        
+        let audioInfo = AudioInfo(duration: UInt64(durationInSeconds * 1000), size: fileSize, mimetype: mimeType)
+        return .success(.audio(audioURL: url, audioInfo: audioInfo))
+    }
+    
+    /// Prepares a file for upload.
+    /// - Parameters:
+    ///   - url: The file URL
+    ///   - type: its UTType
+    ///   - mimeType: the mimeType extracted from the UTType
+    /// - Returns: Returns a `MediaInfo.file` containing the file URL plus the corresponding `FileInfo`
+    private func processFile(at url: URL, mimeType: String?) async -> Result<MediaInfo, MediaUploadingPreprocessorError> {
+        let fileSize = try? UInt64(FileManager.default.sizeForItemAt(url))
+        
+        let fileInfo = FileInfo(mimetype: mimeType, size: fileSize, thumbnailInfo: nil, thumbnailSource: nil)
+        return .success(.file(fileInfo))
+    }
+    
+    // MARK: Images
+    
+    /// Removes the GPS dictionary from an image's metadata
+    /// - Parameters:
+    ///   - url: the URL for the original image
+    ///   - type: its UTType
+    /// - Returns: the URL for the modified image and its size as an `ImageProcessingResult`
+    private func stripLocationFromImage(at url: URL, type: UTType, mimeType: String) async -> Result<ImageProcessingInfo, MediaUploadingPreprocessorError> {
+        guard let originalData = NSData(contentsOf: url),
+              let originalCGImage = UIImage(data: originalData as Data)?.cgImage,
+              let imageSource = CGImageSourceCreateWithData(originalData, nil) else {
```

Comment:
I think these 3 lines could be simplified down to
```suggestion
        guard let imageSource = CGImageSourceCreateWithURL(url as CFURL, nil) else {
```

Reply:
Well it can but I need the `originalCGImage` to read its size and `CGImageDestinationAddImage` it back to the destination